### PR TITLE
🧧 Include and exclude groups for all triggers

### PR DESCRIPTION
**Pull request by rowanseymour**
* except scheduled triggers just yet as they are a different kettle of fish

Captura de Pantalla 2021-06-10 a la(s) 17 06 15
Captura de Pantalla 2021-06-10 a la(s) 17 06 29
Captura de Pantalla 2021-06-10 a la(s) 17 06 37
Captura de Pantalla 2021-06-10 a la(s) 17 06 43
Captura de Pantalla 2021-06-10 a la(s) 17 06 52

and don't even think about trying to include and exclude the same group!

Captura de Pantalla 2021-06-10 a la(s) 17 07 46

### CODE REVIEW

File: temba/triggers/models.py [hunk 1]
```diff
@@ -89,6 +89,7 @@ class Trigger(SmartModel):
     EXPORT_KEYWORD = "keyword"
     EXPORT_FLOW = "flow"
     EXPORT_GROUPS = "groups"
+    EXPORT_EXCLUDE_GROUPS = "exclude_groups"
```
Comment (rowanseymour) [inline temba/triggers/models.py:92]: I feel like adding constants which are literally that string creates more clutter than value but sticking with what is here..

File: temba/triggers/views.py [hunk 2]
```diff
@@ -44,7 +44,31 @@ class BaseTriggerForm(forms.ModelForm):
         widget=SelectWidget(attrs={"placeholder": _("Select a flow"), "searchable": True}),
     )
 
-    conflict_message = _("There already exists a trigger of this type.")
+    groups = TembaMultipleChoiceField(
+        queryset=ContactGroup.user_groups.none(),
+        label=_("Group Filtering"),
```
Comment (rowanseymour) [inline temba/triggers/views.py:49]: @ericnewcomer any ideas why this label isn't appearing? Tearing my hair out...

File: templates/triggers/trigger_create.haml [hunk 3]
```diff
@@ -1,26 +1,11 @@
 -extends "smartmin/base.html"
 -load i18n
 
--block extra-style
```
Comment (rowanseymour) [inline templates/triggers/trigger_create.haml:4]: not used

File: templates/triggers/trigger_create.haml [hunk 4]
```diff
@@ -1,26 +1,11 @@
...
 
   .mb-3.text-xl.mt-2
     -trans "Triggers allow users to start flows based on user actions or schedules."
   
 -block content
   -include "formax.haml"
-
--block extra-script
-  {{ block.super }}
-  <script src="{% url 'django.views.i18n.javascript_catalog' %}"></script>
```
Comment (rowanseymour) [inline templates/triggers/trigger_create.haml:26]: not used

File: temba/triggers/views.py (hunk 2 above)
Comment (rowanseymour) [reply temba/triggers/views.py:49]: oh n/m it's the `"widget_only": True` part

**Review by ericnewcomer — APPROVED** [review]
Nice, I like it.. especially the simplification across trigger types.

**Pull request merged**